homeassistant/components/sense/binary_sensor.py

```diff
@@ -102,12 +103,16 @@ def device_class(self):
         """Return the device class of the binary sensor."""
         return BIN_SENSOR_CLASS
 
-    async def async_update(self):
-        """Retrieve latest state."""
-        from sense_energy.sense_api import SenseAPITimeoutException
-        try:
-            await self._data.update_realtime()
-        except SenseAPITimeoutException:
-            _LOGGER.error("Timeout retrieving data")
-            return
-        self._state = self._name in self._data.active_devices
+    @property
+    def should_poll(self):
+        """Return the deviceshould not poll for updates."""
+        return False
+
+    async def async_added_to_hass(self):
+        """Register callbacks."""
+        @callback
+        def update():
+            """Update the state."""
+            self.async_schedule_update_ha_state(True)
+
+        async_dispatcher_connect(self.hass, SENSE_DEVICE_UPDATE, update)
```

Comment:
The return value of this function is a function that will undo the subscription. Store the return value and call it inside `async_will_remove_from_hass`